### PR TITLE
Closes #208

### LINKED ISSUE #208: CakePHP 3.6 deprecation warnings
Ist anyone working on Code Migration to avoid deprecation warnings popping Up when using this in 3.6.0-betaX? If appreciated I would try to fix These.

### PR DESCRIPTION
**Pull request by swiffer**
Resolves https://github.com/FriendsOfCake/search/issues/208

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -5,3 +5,7 @@
 phpunit.xml.dist export-ignore
 .travis.yml export-ignore
 tests export-ignore
+
+# Git will always convert line endings to LF on checkout.
+# You should use this for files that must keep LF endings, even on Windows.
+* text eol=lf
```
Comment (dereuromark) [inline .gitattributes]: :+1: 
We usually put that on the top though
before the export-ignore part.

**Comment by dereuromark** [comment]
You need to also bump the constraint then:

    "cakephp/cakephp": "^3.4"

is not true anymore after this one.

File: .gitattributes (hunk 1 above)
Comment (swiffer) [reply .gitattributes]: done ;)

File: .gitattributes [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
```
Comment (dereuromark) [inline .gitattributes]: what about the auto part? https://github.com/cakephp/cakephp/blob/master/.gitattributes#L3

**Comment by swiffer** [comment]
i think it's still working on 3.4 at least in 3.5 code works without problems!

**Comment by dereuromark** [comment]
To make sure we should add the same I added to my plugin test matrix: https://github.com/dereuromark/cakephp-geo/blob/master/.travis.yml#L27

**Comment by dereuromark** [comment]
You need 5.7.14 there ar least for phpunit.

**Comment by swiffer** [comment]
@dereuromark ok, I'm done ;)

**Comment by dereuromark** [comment]
Nice work!

File: composer.json [hunk 3]
```diff
@@ -17,17 +17,17 @@
...
-            "role":"Contributor",
-            "homepage":"https://github.com/admad"
+            "name": "ADmad",
+            "role": "Contributor",
+            "homepage": "https://github.com/admad"
         }
     ],
     "require": {
-        "cakephp/cakephp": "^3.4"
+        "cakephp/cakephp": "^3.5"
```
Comment (ADmad) [inline composer.json]: The changes don't work with 3.4?

Comment (swiffer) [reply composer.json]: Sadly getting Errors when running the Testsuite in 3.4.0

**Comment by dereuromark** [comment]
This should be a minor release.
no bc breaks but min requirements raised.

**Comment by ADmad** [comment]
Yup, it will be a minor release due to cake version bump. I'll merge when I am at a PC.

**Comment by ADmad** [comment]
I have merged this in to a `cake-36` branch instead of `master` to prepare for 3.6. Will merge `cake-36` to master once CakePHP 3.6 is released.

**Pull request merged**